sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -139,3 +140,15 @@ def get_next_group_id(self):
 
     self.group_id += 1
     return self.group_id
+
+  # The following functions are used by the users to customize pipeline level configurations
+  # Note: call these function inside the pipeline python function.
+  @staticmethod
+  def set_imagepullsecret(secret_name):
```

Comment:
It's preferable to make this method as an instance method over static, so that it can be chained with other spec setting methods. like: dsl.get_pipeline_spec().set_image_pull_secrets(...).set_active_deadline_seconds(...).set_node_selector(...).

Reply:
The pipeline class, when initially designed, was only used by the compiler. Therefore, not exposing the instance to the authors. That's why I added the static method to fetch the default pipeline instance.

Reply:
I think we already have the get_default_pipeline static method to expose the pipeline instance.

Reply:
Yes, it is not supposed to be used by the authors, though. Technically users can call Pipeline.get_default_pipeline.set_image_pull_secrets if we define the set_image_pull_secrets as a non-static function. However, the concept of default pipeline will be very confusing, then.

Reply:
How about a class PipelineConf that takes all the configuration and expose a module level function to assign the PipelineConf to the Pipeline. Mixing the user functions and internally used functions inside the Pipeline does not seem to be an elegant way. WDYT?

Reply:
sgtm if it looks like: `dsl.pipeline_spec().set_image_pull_secrets(...)`.

Reply:
Make sense. 
Done.